guide_software_engineer.md

```diff
@@ -21,16 +21,17 @@ Souzoh の企業情報やイベント情報、カジュアル面談の申込方
-| Coding Test             | N/A                                   | 1 時間        | N/A                        | ソフトウェアエンジニアとしての基礎力を確かめるため            | 外部サービスを利用し、基礎的なアルゴリズムのテストに取り組んでいただきます       |
-| 書類選考                | N/A                                   | N/A           | 上記と合わせて 1〜5 営業日 | 募集要項の必須条件を満たしているかを確認するため              | Coding Test の結果とあわせて書類選考を実施します                                 |
-| System Design Interview | ソフトウェアエンジニア/EM/CTO 1〜2 名 | 1 時間        | 1〜5 営業日                | 「問題解決能力」を確認するため                                | システム設計に関するトピックについて、応募者と面接官でディスカッションを行います |
-| 面接                    | EM 1 名                               | 30 分〜1 時間 | 1〜5 営業日                | Souzoh で働くにあたってミスマッチがないかお互いに確認するため | Engineering Manager と面接をしていただきます                                     |
-| 面接                    | CTO                                   | 30 分〜1 時間 | 1〜5 営業日                | Souzoh で働くにあたってミスマッチがないかお互いに確認するため | CTO と面接をしていただきます                                                     |
+| 選考                    | 対象者  | 出席者                                | 時間          | 結果連絡までの日数         | 目的                                                          | 内容                                                                             |
+| ----------------------- | ------- | ------------------------------------- | ------------- | -------------------------- | ------------------------------------------------------------- | -------------------------------------------------------------------------------- |
+| Coding Test             | ML 以外 | N/A                                   | 1 時間        | N/A                        | ソフトウェアエンジニアとしての基礎力を確かめるため            | 外部サービスを利用し、基礎的なアルゴリズムのテストに取り組んでいただきます       |
+| 書類選考                | 全員    | N/A                                   | N/A           | 1〜5 営業日                | 募集要項の必須条件を満たしているかを確認するため              | Coding Test の結果とあわせて書類選考を実施します（ML エンジニアは書類のみ）      |
+| ML 選考課題             | ML      | N/A                                   | 数日程度      | 1〜5 営業日                | ML モデル作成に関わる実装能力を確認するため                   | ML モデル作成に関する課題に取り組んでいただきます                                |
+| System Design Interview | 全員    | ソフトウェアエンジニア/EM/CTO 1〜2 名 | 1 時間        | 1〜5 営業日                | 「問題解決能力」を確認するため                                | システム設計に関するトピックについて、応募者と面接官でディスカッションを行います |
+| 面接                    | 全員    | EM 1 名                               | 30 分〜1 時間 | 1〜5 営業日                | Souzoh で働くにあたってミスマッチがないかお互いに確認するため | Engineering Manager と面接をしていただきます                                     |
+| 面接                    | 全員    | CTO                                   | 30 分〜1 時間 | 1〜5 営業日                | Souzoh で働くにあたってミスマッチがないかお互いに確認するため | CTO と面接をしていただきます                                                     |
```

Comment:
How about separating this section into ML and others?
IMO, it will be easier to read and understand.